src/webview/html/messageHeaderAsHtml.js

```diff
@@ -29,22 +31,23 @@ export default (
     return '';
   }
 
-  if (isStreamNarrow(narrow)) {
+  if (isStreamNarrow(narrow) || isTopicNarrow(narrow)) {
     const topicNarrowStr = JSON.stringify(topicNarrow(item.display_recipient, item.subject));
     const topicHtml = renderSubject(item);
 
     return template`
 <div
-  class="header-wrapper header topic-text"
+  class="header-wrapper header topic-content"
   data-narrow="${topicNarrowStr}"
   data-msg-id="${item.id}"
 >
-  $!${topicHtml}
+  <div class="topic-text">$!${topicHtml}</div>
+  <div class="human-date">$!${humanDate(new Date(item.timestamp * 1000))}</div>
 </div>
     `;
   }
 
-  if (item.type === 'stream') {
+  if (item.type === 'stream' && (isSpecialNarrow(narrow) || isHomeNarrow(narrow))) {
```

Comment:
Is this in fact redundant? (And in the `private` case below, is the new condition in fact equivalent to the old?) How do you know -- or, how would you find out for sure?

Also, there's already kind of a long list of possible types of narrows. What if a new one gets added -- how can you write this so that it will *stay* correct then?

I think what you probably want here is `!isStreamOrTopicNarrow`. That's what this block is really about -- that's the description that will stay true regardless of what other miscellaneous types of narrows exist.

Another solution which would make sure you've comprehensively covered the cases, and make sure it stays correct if/when more cases are added, would be to use `caseNarrow`. (See the example uses of it in `utils/narrow.js`. See also `git log --stat -p fa6134aa6^..e9fe1e801` .) But I think in this case the simpler `isStreamOrTopicNarrow` will do just fine.

Reply:
Oh, or here's an alternate way of organizing this logic: put a `caseNarrow` at the *outside*, before looking at `item.type`. Then you get to be super explicit about what the behavior is for each possible type of narrow.

Because much of the logic will be shared across several types of narrow, I'd probably structure that something like this:
```
  const headerStyle = caseNarrow(narrow, {
    stream: () => 'topic+date',
    pm: () => 'date',
    search: () => 'full',
    // ... etc.    
  });

  // Now we're completely done with `narrow` -- the information we need from it is all in `headerStyle`.

  if (item.type === 'stream' && headerStyle === 'topic+date') {
    // ...
  } else // ...
```

The exact set of possible values for `headerStyle` will be up to what you're doing, and might change in later commits in the branch. This will be a way to separate that decision-making in its own bit of code where it's easy to see what's happening.

Reply:
Thanks for the detailed analysis!

I agree with all of the points you made. I do think the second solution is what we would want to implement. Unfortunately I did see the review on this commit after I saw the others, so I'll take some time to implement this (and fix conflicts) and push the final version.